include/onnxruntime/core/providers/coreml/coreml_provider_factory.h

```diff
@@ -31,9 +31,11 @@ enum COREMLFlags {
   // Create an MLProgram. By default it will create a NeuralNetwork model. Requires Core ML 5 or later.
   COREML_FLAG_CREATE_MLPROGRAM = 0x010,
 
+  // Exclude ANE as sometimes this decrease performance
+  COREML_FLAG_USE_CPUAndGPU = 0x020,
```

Comment:
nit: a name like `COREML_FLAG_USE_CPU_AND_GPU_ONLY` would be more consistent with the other names.

Reply:
with this new flag, what does it mean if a user specifies both `COREML_FLAG_USE_CPU_ONLY` and `COREML_FLAG_USE_CPUAndGPU`? should we treat that as an error?

Reply:
CoreMLtools have the name as CPUAndGPU CPUAndANE CPUONLY ALL
https://github.com/apple/coremltools/blob/275c7e2c3ac44a23cd013a8b8a5f90c1a9991a46/coremlpython/CoreMLPython.mm#L144

Should we align the name with them？

Reply:
"CPU_AND_GPU ONLY " sounds a little weird. "CPU ONLY", "GPU ONLY" is more meanningful.

Reply:
> with this new flag, what does it mean if a user specifies both `COREML_FLAG_USE_CPU_ONLY` and `COREML_FLAG_USE_CPUAndGPU`? should we treat that as an error?

Yeah, I think so. Will fix it.

Reply:
aligning with the coreml names seems reasonable.